packages/xo-server/src/xo-mixins/xen-servers.js

```diff
@@ -390,15 +405,17 @@ export default class {
   }
 
   async disconnectXenServer(id) {
-    const xapi = this._xapis[id]
-    if (!xapi) {
-      throw noSuchObject(id, 'xenServer')
+    const status = this._getXenServerStatus(id)
+    if (status === 'disconnected') {
+      return
     }
 
+    const xapi = this._xapis[id]
     delete this._xapis[id]
-    delete this._serverIdsByPool[xapi.pool.$id]
-
-    xapi.xo.uninstall()
+    if (status === 'connected') {
+      delete this._serverIdsByPool[xapi.pool.$id]
+      xapi.xo.uninstall()
+    }
     return xapi.disconnect()
```

Comment:
What will this do if the server is "connecting"?

Reply:
Nothing because the session is not yet established.

Reply:
But if the session is established it will close the connection and is the behaviour that we want.